internal/pkg/gateway/registry.go

```diff
@@ -348,7 +341,14 @@ func (reg *registry) registerChannel(channel string) error {
 			}
 		}
 	}
-	reg.channelsInitialized[channel] = true
+	reg.endorsersInitialized[channel] = true
 
 	return nil
 }
+
+func (reg *registry) configUpdate(bundle *channelconfig.Bundle) {
```

Comment:
1. The goroutine that calls the callback writes to reg.configValid but the goroutine that reads it in line 282 in the registry, does not see that write, because there is no memoryhappens before relation. You need to lock here with the configLock.
2. Why do you invalidate the entire map if only a specific channel was updated? You can extract the channel name from the bundle.

Reply:
Thanks, I hadn't spotted how to extract the channel name.  I've made that change and switched to a sync.Map so that the callback won't get potentially blocked by the registry making peer/orderer connections

Reply:
See my [comment](https://github.com/hyperledger/fabric/pull/2927#discussion_r710454334) about that problem.